src/contracts/batching/mod.rs

```diff
@@ -57,27 +54,31 @@ impl IdentityManager for Contract {
             "Connected to the WorldID Identity Manager"
         );
 
-        let sitter = Sitter::new(ethereum.clone()).await?;
+        let initial_leaf_value = options.initial_leaf_value;
+        let tree_depth = options.tree_depth;
 
         let identity_manager = Self {
             ethereum,
-            sitter,
             abi,
+            initial_leaf_value,
+            tree_depth,
         };
 
         Ok(identity_manager)
     }
 
     fn tree_depth(&self) -> usize {
-        todo!()
+        self.tree_depth
     }
 
     fn initial_leaf_value(&self) -> Field {
-        todo!()
+        self.initial_leaf_value
     }
 
     fn group_id(&self) -> U256 {
```

Comment:
Just remove this altogether

Reply:
It'll be removed in the next PR where I pull out the legacy contract entirely.